src/blop/objectives.py

```diff
@@ -143,16 +143,56 @@ def __init__(
                 raise ValueError("'target' must be either 'min', 'max', a number, or a tuple of numbers.")
 
     @property
-    def domain(self) -> tuple[float, float]:
+    def search_domain(self) -> tuple[float, float] | set[int] | set[str] | set[bool]:
+        return self._search_domain
+
+    @search_domain.setter
+    def search_domain(self, value: tuple[float, float] | set[int] | set[str] | set[bool]):
+        """
+        Make sure that the search domain is within the trust domain before setting it.
+        """
+        value = validate_set(value, type=self.type)
+        trust_domain = self.trust_domain
+        if is_subset(value, trust_domain, type=self.type, proper=False):
+            self._search_domain = cast(tuple[float, float] | set[int] | set[str] | set[bool], value)
```

Comment:
Cast does nothing here. The type checker already knows the type from the method signature.
```suggestion
            self._search_domain = value
```